pkg/apis/resource/validation/validation.go

```diff
@@ -235,10 +316,22 @@ func validateDeviceClaimConfiguration(config resource.DeviceClaimConfiguration,
 	return allErrs
 }
 
-func validateRequestNameRef(name string, fldPath *field.Path, requestNames sets.Set[string]) field.ErrorList {
-	allErrs := validateRequestName(name, fldPath)
+func validateRequestNameRef(name string, fldPath *field.Path, requestNames requestNames) field.ErrorList {
+	var allErrs field.ErrorList
+	segments := strings.Split(name, "/")
+	if len(segments) > 2 {
+		allErrs = append(allErrs, field.Invalid(fldPath, name, "must be the name of a request in the claim or the name of a request and a subrequest separated by '/'"))
+		return allErrs
+	}
+	// Not sure if this is the best way to handle this validation. If can create
+	// too separate error messages if both the references to the request and the
+	// subrequest fails validation. Do we even need it? Validating that there
+	// is a corresponding request should be sufficient.
```

Comment:
I think one can argue both ways.

I prefer duplicating the check. Otherwise a user submits a claim with an invalid request name and a valid request reference, fixes the name and forgets to update the reference, and then gets another error in the next attempt. Better list all problematic fields at once.